### PR TITLE
[Flax] Fix unet and ddim scheduler

### PR DESCRIPTION
**Pull request by patrickvonplaten**
```python
from diffusers import FlaxStableDiffusionPipeline
from diffusers import StableDiffusionPipeline
from jax import pmap
import numpy as np
import jax
import jax.numpy as jnp
import torch
from flax.jax_utils import replicate
from flax.training.common_utils import shard


def safety_checker(images=None, clip_input=None):
    return images, False


prng_seed = jax.random.PRNGKey(0)
num_inference_steps = 2

pipeline_torch = StableDiffusionPipeline.from_pretrained("hf-internal-testing/tiny-stable-diffusion-pipe")
pipeline_torch.safety_checker = safety_checker
pipeline, params = FlaxStableDiffusionPipeline.from_pretrained("hf-internal-testing/tiny-stable-diffusion-pipe")

latents_torch = torch.rand(1, 4, pipeline_torch.unet.config.sample_size, pipeline_torch.unet.config.sample_size)
latents = jnp.array(latents_torch.cpu())

prompt = "A cinematic film still of Morgan Freeman starring as Jimi Hendrix, portrait, 40mm lens, shallow depth of field, close up, split lighting, cinematic"

num_samples = 1

prompt_ids = pipeline.prepare_inputs(prompt)
images_np = pipeline(prompt_ids, params, prng_seed, num_inference_steps, latents=latents, debug=True).images
images = pipeline.numpy_to_pil(np.asarray(images_np.reshape((num_samples,) + images_np.shape[-3:])))
print(100 * "-")
images_torch_np = pipeline_torch(prompt, latents=latents_torch, height=256, width=256, num_inference_steps=num_inference_steps, output_type="np").images
images_torch = pipeline_torch.numpy_to_pil(images_torch_np)

print(f"Diff Sum: {np.abs(images_torch_np - images_np).sum()}")
print(f"Diff Max: {np.abs(images_torch_np - images_np).max()}")
import ipdb; ipdb.set_trace()
print("works!")
```

Now gives:
```
Diff Sum: 0.19847372174263                                                                                             │
Diff Max: 0.00012359023094177246 
```

### CODE REVIEW

File: src/diffusers/models/embeddings_flax.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def get_sinusoidal_embeddings(timesteps, embedding_dim):
     embeddings. :return: an [N x dim] tensor of positional embeddings.
     """
     half_dim = embedding_dim // 2
-    emb = math.log(10000) / (half_dim - 1)
+    emb = math.log(10000) / (half_dim - freq_shift)
```
Comment (patrickvonplaten) [inline src/diffusers/models/embeddings_flax.py:32]: @patil-suraj @younesbelkada -> this is why we only got `4e-3` tolerance. We should now get something closer to `1e-4` tolerance.

Comment (younesbelkada) [reply src/diffusers/models/embeddings_flax.py:32]: Very nice !! Thanks for spotting the conversion nit 🚀

Comment (younesbelkada) [reply src/diffusers/models/embeddings_flax.py:32]: Just tried your fix and the test pass with
`torch.testing.assert_allclose(converted_flax_output, torch_output, rtol=1e-04, atol=1e-04)` 
Thanks again!

File: src/diffusers/schedulers/scheduling_ddim_flax.py [hunk 2]
```diff
@@ -216,6 +216,9 @@ def step(
         # - pred_sample_direction -> "direction pointing to x_t"
         # - pred_prev_sample -> "x_t-1"
 
+        # TODO(Patrick) - eta is always 0.0 for now, allow to be set in step function
+        eta = 0.0
```
Comment (jonatanklosko) [inline src/diffusers/schedulers/scheduling_ddim_flax.py:220]: Rather than a step argument, maybe it would make more sense as a configuration property? Any additional arguments to `step` lead to this:

https://github.com/huggingface/diffusers/blob/ce31f83d8c44c977dcf8413c99b8a6cb7f189c07/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L242-L249

Comment (patrickvonplaten) [reply src/diffusers/schedulers/scheduling_ddim_flax.py:220]: This one I think we can solve by just letting the user forward `**step_kwargs` . It should IMO not be in the config because `eta` is a parameter that one might want to change between forward passes (contrary to `offset_steps`)
